compat/test/browser/suspense-hydration.test.js

```diff
@@ -164,6 +216,196 @@ describe('suspense hydration', () => {
+		hydrate(
+			<ul>
+				<Fragment>
+					<li onClick={listeners[0]}>0</li>
+					<li onClick={listeners[1]}>1</li>
+				</Fragment>
+				<Suspense>
+					<Lazy />
+				</Suspense>
+				<Fragment>
+					<li onClick={listeners[3]}>3</li>
+					<li onClick={listeners[4]}>4</li>
+				</Fragment>
+			</ul>,
+			scratch
+		);
+		rerender(); // Flush rerender queue to mimic what preact will really do
+		expect(scratch.innerHTML).to.equal(originalHtml);
+		expect(getLog()).to.deep.equal([]);
+		expect(listeners[4]).not.to.have.been.called;
+
+		clearLog();
+		scratch.querySelector('li:last-child').dispatchEvent(createEvent('click'));
+		expect(listeners[4]).to.have.been.calledOnce;
+
+		return resolve(() => (
+			<Fragment>
+				<li onClick={listeners[2]}>2</li>
+			</Fragment>
+		)).then(() => {
+			rerender();
+			expect(scratch.innerHTML).to.equal(originalHtml);
+			expect(getLog()).to.deep.equal([]);
+			clearLog();
+
+			scratch
+				.querySelector('li:nth-child(3)')
+				.dispatchEvent(createEvent('click'));
+			expect(listeners[2]).to.have.been.calledOnce;
+
+			scratch
+				.querySelector('li:last-child')
+				.dispatchEvent(createEvent('click'));
+			expect(listeners[4]).to.have.been.calledTwice;
+		});
+	});
+
+	it('should properly hydrate suspense with Component & Fragment siblings', () => {
+		const originalHtml = ul([li(0), li(1), li(2), li(3), li(4)].join(''));
+
+		const listeners = [
+			sinon.spy(),
+			sinon.spy(),
+			sinon.spy(),
+			sinon.spy(),
+			sinon.spy()
+		];
+
+		scratch.innerHTML = originalHtml;
+		clearLog();
+
+		const [Lazy, resolve] = createLazy();
+		hydrate(
+			<List>
+				<Fragment>
+					<ListItem onClick={listeners[0]}>0</ListItem>
+					<ListItem onClick={listeners[1]}>1</ListItem>
+				</Fragment>
+				<Suspense>
+					<Lazy />
+				</Suspense>
+				<Fragment>
+					<ListItem onClick={listeners[3]}>3</ListItem>
+					<ListItem onClick={listeners[4]}>4</ListItem>
+				</Fragment>
+			</List>,
+			scratch
+		);
+		rerender(); // Flush rerender queue to mimic what preact will really do
+		expect(scratch.innerHTML).to.equal(originalHtml);
+		expect(getLog()).to.deep.equal([]);
+		expect(listeners[4]).not.to.have.been.called;
+
+		clearLog();
+		scratch.querySelector('li:last-child').dispatchEvent(createEvent('click'));
+		expect(listeners[4]).to.have.been.calledOnce;
+
+		return resolve(() => (
+			<Fragment>
+				<ListItem onClick={listeners[2]}>2</ListItem>
+			</Fragment>
+		)).then(() => {
+			rerender();
+			expect(scratch.innerHTML).to.equal(originalHtml);
+			expect(getLog()).to.deep.equal([]);
+			clearLog();
+
+			scratch
+				.querySelector('li:nth-child(3)')
+				.dispatchEvent(createEvent('click'));
+			expect(listeners[2]).to.have.been.calledOnce;
+
+			scratch
+				.querySelector('li:last-child')
+				.dispatchEvent(createEvent('click'));
+			expect(listeners[4]).to.have.been.calledTwice;
+		});
+	});
+
+	it.skip('should properly hydrate suspense when resolves to a Fragment', () => {
```

Comment:
Currently not supported but I didn't realize that until after I wrote the tests so since I wrote it I figured I'd add it so that maybe one day we can enable it.